docs/apps/datahub.md

```diff
@@ -7,3 +7,17 @@ outline: deep
 ## Chapter 1
 
 ## Chapter 2
+
+## Metadata pages
+
+Each metadata has its own detailed page, made of multiple sections, described as followed.
+
+### Keywords
+
+The keyword section is dynamically generated from the `metadata-info` component. The metadata comes from the harvested `record.metadata` object, which contains a `keywords` array or strings. Through a test on the array's length, if the array is empty, the section will not be displayed.
+
+### Usage and constraints
+
+The usage and constraints section is dynamically generated from the `metadata-info` component. The metadata comes from the harvested `record.metadata` object, which contains a `constraints` array or strings. Both the usage and constraints are stored in this array. Through a test on the array's length (`hasUsage`), if the array is empty, the section will not be displayed.
+
+A part of the section can however, in some cases, be displayed despite an empty array. In this scenario, a message will be displayed `record.metadata.noUsage` to warn the user.
```

Comment:
Thanks for adding some doc. I think we don't need to go so much into implementation details here and name variables, for example. This will be hard to keep up to date and if developers want to know such detail, they will take a look at the code IMO.

Maybe we could just list the different sections for now or possibly give some information on what metadata attributes they are based on.

If we want to point to some code, the https://github.com/geonetwork/geonetwork-ui/tree/main/libs/ui/elements/src/lib/metadata-info component might be worth linking.

What do you think?

Reply:
Yes, I agree. I wasn't really sure of what to write, but your idea seems more concise. Will edit.